pkg_resources/__init__.py

```diff
@@ -2911,7 +2911,7 @@ def __init__(
         platform: str | None = None,
         precedence: int = EGG_DIST,
     ) -> None:
-        self.project_name = safe_name(project_name or 'Unknown')
+        self.project_name = safe_name(project_name or 'Unknown').replace('.', '-')
```

Comment:
I don't think we should change `project_name`.
If I am not mistaken in all the discussions we reached the conclusion that the "publicly facing" name of projects should stay as given by the users (e.g. with `.`), the "canonicalised/normalised" name would be only a aspect of "machine driven" process, e.g. dependency resolution, file naming etc. That is also why PyPA does not require normalisation of metadata.